pyomo/contrib/alternative_solutions/shifted_lp.py

```diff
@@ -0,0 +1,225 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2024
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+import pyomo.environ as pe
+from pyomo.common.collections import ComponentMap
+from pyomo.gdp.util import clone_without_expression_components
+from pyomo.contrib.fbbt.fbbt import compute_bounds_on_expr
+from pyomo.contrib.alternative_solutions import aos_utils
+
+
+def _get_unique_name(collection, name):
+    """Create a unique name for an item that will be added to a collection."""
+    if name not in collection:
+        return name
+    else:
+        i = 1
+        while "{}_{}".format(name, i) not in collection:
+            i += 1
+        return "{}_{}".format(name, i)
+
+
+def _set_slack_ub(expression, slack_var):
+    """
+    Use FBBT to compute an upper bound for a slack variable on an equality
+    expression."""
+    slack_lb, slack_ub = compute_bounds_on_expr(expression)
+    assert slack_ub >= 0
+    slack_var.setub(slack_ub)
+
+
+def get_shifted_linear_model(model, block=None):
+    """
+    Converts an (MI)LP with bounded (discrete and) continuous variables
+    (l <= x <= u) into a standard form where where all continuous variables
+    are non-negative reals and all constraints are equalities. For a pure LP of
+    the form,
+
+    min/max cx
+    s.t.
+        A_1 * x  =  b_1
+        A_2 * x <= b_2
+        l <= x <= u
+
+    a problem of the form,
+
+    min/max c'z
+    s.t.
+        Bz = q
+        z >= 0
```

Comment:
Not necessary for this PR, but in the future this could be modified to use the `compile_standard_form` writer from `repn.plugins.standard_form`.

Reply:
Good idea!